scalyr_agent/builtin_monitors/tests/kubernetes_monitor_test.py

```diff
@@ -194,18 +194,61 @@ def test_basic_case(self):
 
         self.assertTrue(warmer.is_warm(self.NAMESPACE_1, self.POD_1))
 
-    def test_already_warm(self):
+    def test_already_warm_before_marking(self):
         warmer = self.__warmer_test_instance
         fake_cache = self.__fake_cache
 
+        # Round 1: WarmingEntry does not exist case
         warmer.begin_marking()
+        fake_cache.simulate_add_pod_to_cache(self.NAMESPACE_1, self.POD_1)
         warmer.mark_to_warm(self.CONTAINER_1, self.NAMESPACE_1, self.POD_1)
         warmer.end_marking()
 
+        fake_cache.set_response(self.NAMESPACE_1, self.POD_1, success=True)
+        warmer.block_until_idle(self.timeout)
+
+        stats = warmer.get_report_stats()
+        success_count = stats.get('success', (1, 1))
+        self.assertEqual(success_count[0], 0)
+        already_warm_count = stats.get('already_warm', (0, 0))
+        self.assertEqual(already_warm_count[0], 1)
```

Comment:
What's the point of round 1?  In my mind, you do not need to simulate the entry being retrieved from the API to test the `already_warm` case.

Reply:
Round 1 was for testing the above "new entry" code branch